resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesTestComponents.scala

```diff
@@ -106,7 +106,7 @@ private[spark] object SparkAppLauncher extends Logging {
     val sparkSubmitExecutable = sparkHomeDir.resolve(Paths.get("bin", "spark-submit"))
     logInfo(s"Launching a spark app with arguments $appArguments and conf $appConf")
     val appArgsArray =
-      if (appArguments.appArgs.length > 0) Array(appArguments.appArgs.mkString(" "))
```

Comment:
Doesn't this make `appArgsArray` basically redundant?

Reply:
Yes unless you pass a null... I updated my comment above ^^.